### PR TITLE
feat: add juju 3.1 support

### PR DESCRIPTION
**Pull request by mert-kirpici**
New branch: `3.1` created for 3.1 series. he new track is [requested](https://forum.snapcraft.io/t/track-request-for-prometheus-juju-exporter-snap/36734) however not granted yet, so the publish workflow will fail, however we can trigger it manually whenever we are granted the track.

### CODE REVIEW

**Review by esunar — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ asyncio
 
 # Pinning the last release that was
 # natively designed for Juju 2.9
```
Comment (agileshaw) [inline requirements.txt:6]: The comment here needs to be removed

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ name: PR workflow running lint checkers, unit and functional tests
 on:
   pull_request:
     types: [ opened, synchronize, reopened ]
-    branches: [ master, main ]
+    branches: [ 3.1 ]
```
Comment (agileshaw) [inline .github/workflows/pr.yaml:10]: Based on the conclusion from previous discussion (when we were laying out branching strategy initially), we will continue using mater branch as the default branch and the target for merging pull request. Every time we push new changes to main, we sync the changes to the latest supporting branch (in this case, 3.1) and the release workflow will handle the rest. The philosophy is similar to what's in python-libjuju: develop on the main and merge the changes to version branches).

**Review by agileshaw — CHANGES_REQUESTED** [review]
The pull request should be merged to `main` instead of `3.1` branch as explained in-line. Also, could you please update the README ([this part](https://github.com/canonical/prometheus-juju-exporter#juju-version-compatibility)) accordingly?

File: requirements.txt (hunk 1 above)
Comment (mert-kirpici) [reply requirements.txt]: done.

File: .github/workflows/pr.yaml (hunk 2 above)
Comment (mert-kirpici) [reply .github/workflows/pr.yaml]: changed the base branch to main. i was not aware of that previous discussion.

Comment (agileshaw) [reply .github/workflows/pr.yaml]: Thanks. Yes we could've documented this better. I will open a PR afterwards to add the info in README.

**Review by agileshaw — APPROVED** [review]

**Pull request merged**